framework/SupervisedLearning/SupervisedLearning.py

```diff
@@ -124,6 +124,7 @@ def __init__(self):
     self.target = None             # "outputs" of this model
     self.amITrained = False        # "True" if the ROM is alread trained
     self._dynamicHandling = False  # time-like dependence in the model?
+    self.dynamicFeatures = False   # time-like dependence in the feature space? FIXME: this is not the right design
```

Comment:
I think so. Thats why I asked (if the new API already implemented this) before adding this modification.
I think the ROM should use the dataset directly.

Reply:
There's a flag in place to specify whether training should be done on a dictionary (undesirable, but legacy) or on the dataset directly (desirable but new).

This flag exists on the SupervisedLearning inheritor and is called:
```python
self.needsDictTraining = False    # we train directly on the dataset
```

It's similar to the features added in this PR but more generally passes the dataset rather than reshaping data.

Reply:
Sorry. I did not know that this feature was available already. I will try this out tonight after work.

Reply:
Thanks @aalfonsi

Reply:
@PaulTalbot-INL @wangcj05 I do not find this flag. Is it in a branch?

Reply:
@aalfonsi I do not find it either. @PaulTalbot-INL Is it used in your branch for SONAR?

Reply:
@wangcj05 I checked his branch in his fork. Yes it is in there. Let me know what you want me to do.

Reply:
@aalfonsi I have asked Paul to comment on this. It seems to me it is probably ok to accept your changes without the self.needsDictTraining.